### PR TITLE
fix(warn): avoid redundant usage warn

### PR DESCRIPTION
**Pull request by leemove**
fix: https://github.com/vuejs/router/issues/1306

### CODE REVIEW

**Review by posva — CHANGES_REQUESTED** [review]
Can you add a test?

**Comment by leemove** [comment]
> Can you add a test?

done

File: packages/router/__tests__/RouterView.spec.ts [hunk 1]
```diff
@@ -411,6 +411,35 @@ describe('RouterView', () => {
       expect('can no longer be used directly inside').toHaveBeenWarned()
     })
 
+    it('does not warn if router-view is not used directly inside transition', async () => {
```
Comment (posva) [inline packages/router/__tests__/RouterView.spec.ts:414]:
```suggestion
    it('does not warn if RouterView is not a direct-child of transition', async () => {
```

File: packages/router/src/RouterView.ts [hunk 2]
```diff
@@ -240,9 +241,12 @@ export const RouterView = RouterViewImpl as unknown as {
 function warnDeprecatedUsage() {
   const instance = getCurrentInstance()!
   const parentName = instance.parent && instance.parent.type.name
+  const parentSubTreeType = instance.parent?.subTree?.type
```
Comment (posva) [inline packages/router/src/RouterView.ts:244]:
```suggestion
  const parentSubTreeType = instance.parent && instance.parent.subTree && instance.parent.subTree.type
```

**Review by posva — APPROVED** [review]
Thanks! I will give this a local check later on and merge

**Pull request merged**